sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalog.scala

```diff
@@ -184,4 +184,9 @@ abstract class ExternalCatalog {
 
   def listFunctions(db: String, pattern: String): Seq[String]
 
+  // --------------------------------------------------------------------------
+  // Resources
+  // --------------------------------------------------------------------------
+
+  def addJar(path: String): Unit
```

Comment:
Add a resource to the underlying Hive metastore for DDL operations

For example, if we do not use HiveClient to pass the `ADD JAR` command to Hive metastore, we are unable to create the table. Thus, it sounds fine to put `addJar` into `ExternalCatalog`.
```Scala
    val testJar = TestHive.getHiveFile("hive-hcatalog-core-0.13.1.jar").getCanonicalPath
    sql(s"ADD JAR $testJar")
    sql(
      """
        |CREATE TABLE t1(a string, b string)
        |ROW FORMAT SERDE 'org.apache.hive.hcatalog.data.JsonSerDe'
      """.stripMargin)
```